Nodejs/Product/Npm/PackageJsonFactory.cs

```diff
@@ -1,14 +1,71 @@
-// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
 
+using System;
+using System.IO;
+using System.Threading;
 using Microsoft.NodejsTools.Npm.SPI;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
 
 namespace Microsoft.NodejsTools.Npm
 {
-    public class PackageJsonFactory
+    public static class PackageJsonFactory
     {
-        public static IPackageJson Create(IPackageJsonSource source)
+        public static IPackageJson Create(string fullPathToFile)
         {
-            return source.Package == null ? null : new PackageJson(source.Package);
+            if (fullPathToFile == null || !fullPathToFile.EndsWith("package.json", StringComparison.OrdinalIgnoreCase))
+            {
+                throw new ArgumentException("Expected full path to 'package.json' file.", nameof(fullPathToFile));
+            }
+
+            if (File.Exists(fullPathToFile))
+            {
+                var retryInterval = 500;
+                var attempts = 5;
+
+                // populate _source with retries for recoverable errors.
+                while (--attempts >= 0)
+                {
+                    try
+                    {
+                        using (var fin = new FileStream(fullPathToFile, FileMode.Open, FileAccess.Read, FileShare.Read))
+                        using (var reader = new StreamReader(fin))
+                        {
+                            object source = null;
+
+                            var text = reader.ReadToEnd();
```

Comment:
If we're just going to read the whole file, would it be easier to call `File.ReadAllText`?

Reply:
`File.ReadAllText` doesn't seem to set `FileMode`, `FileAccess`, or `FileShare`.

Reply:
1) I'm not that concerned, since it lets go when it's done.
2) I'm pretty sure it does.